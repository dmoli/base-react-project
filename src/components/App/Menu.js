import React, { Component } from 'react';
import Link from 'next/link';
import { FormattedMessage } from 'react-intl';

class Menu extends Component {
  constructor() {
    super();
    this.state = {
      open: false,
    };
    this.handleOpenMenu = this.handleOpenMenu.bind(this);
  }

  handleOpenMenu() {
    let open = true;
    if (this.state.open) {
      open = false;
    }
    this.setState({ open });
  }

  render() {
    return (
      <div>
        <span className='burger-button icon-menu' onClick={this.handleOpenMenu}></span>
        <nav className={this.state.open ? 'menu active' : 'menu'}>
          <ul>
            <li>
              <Link href='/'>
                <a>
                  <FormattedMessage
                    id='link.home'
                    defaultMessage='Hi link home'
                  />
                </a>
              </Link>
            </li>
            <li>
              <Link href='/post'>
                <a>
                  <FormattedMessage
                    id='link.post'
                    defaultMessage='Hi link post'
                  />
                </a>
              </Link>
            </li>
            <li>
              <Link href='/rebass'>
                <a>
                  <FormattedMessage
                    id='link.rebass'
                    defaultMessage='Hi link rebass'
                  />
                </a>
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    );
  }
}

export default Menu;
