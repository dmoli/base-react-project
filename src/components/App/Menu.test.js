import React from 'react';
import renderer from 'react-test-renderer';

import Menu from './Menu';
import createComponentWithIntl from '../../scripts/helperWithIntl';

test('Menu rendered correctly', () => {
  const component = createComponentWithIntl(<Menu />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
