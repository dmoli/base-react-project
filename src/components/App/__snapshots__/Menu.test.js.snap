// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu rendered correctly 1`] = `
<div>
  <span
    className="icon-menu sc-bdVaJa jFQTqy"
    onClick={[Function]}
  />
  <nav
    className="sc-bwzfXH gssRyn"
    open={false}
  >
    <ul
      className="sc-htpNat cuBxGo"
    >
      <li
        className="sc-bxivhb gLOCki"
      >
        <a
          className="sc-ifAKCX hOzByA"
          onClick={[Function]}
        >
          <span>
            Hi link home
          </span>
        </a>
      </li>
      <li
        className="sc-bxivhb gLOCki"
      >
        <a
          className="sc-ifAKCX hOzByA"
          onClick={[Function]}
        >
          <span>
            Hi link post
          </span>
        </a>
      </li>
      <li
        className="sc-bxivhb gLOCki"
      >
        <a
          className="sc-ifAKCX hOzByA"
          onClick={[Function]}
        >
          <span>
            Hi link rebass
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
